mypy/semanal_namedtuple.py

```diff
@@ -646,10 +646,17 @@ def save_namedtuple_body(self, named_tuple_info: TypeInfo) -> Iterator[None]:
                 if isinstance(sym.node, (FuncBase, Decorator)) and not sym.plugin_generated:
                     # Keep user-defined methods as is.
                     continue
+                elif isinstance(sym.node, Var):
+                    # NamedTuple fields are initialized in the generated __init__.
+                    sym.node.is_uninitialized = False
+                    pass
                 # Keep existing (user-provided) definitions under mangled names, so they
                 # get semantically analyzed.
                 r_key = get_unique_redefinition_name(key, named_tuple_info.names)
                 named_tuple_info.names[r_key] = sym
+            if isinstance(value.node, Var):
+                # NamedTuple fields are initialized in the generated __init__.
+                value.node.is_uninitialized = False
```

Comment:
I honestly don't understand why I needed to add this check _twice_, but if I don't, then either the self-check or the tests fails.